wire-library/wire-runtime-swift/src/test/swift/CodableTests.swift

```diff
@@ -333,6 +333,53 @@ extension CodableTests {
         try assertDecode(json: json, expected: proto)
         try assertEncode(proto: proto, expected: json)
     }
+
+    func testNegativeDurationRoundtrip() throws {
+        let json = """
+        {"duration":"-1.900s"}
+        """
+
+        let proto = DurationAndTimestamp(
+            duration: Wire.Duration(seconds: -2, nanos: 100000000)
+        )
+
+        try assertDecode(json: json, expected: proto)
+        try assertEncode(proto: proto, expected: json)
+    }
+
+    func testLargeDurationRoundtrip() throws {
+        let json = """
+        {
+          "duration":"18014398509481984s",
+          "timestamp":"4001-01-01T00:00:00Z"
+        }
+        """
+        let proto = DurationAndTimestamp(
+            duration: Duration(seconds: 1 << 54, nanos: 0),
+            timestamp: Timestamp(date: .distantFuture)
+        )
+
+        // In theory, we should use a much further distant future; however, there woudl be rounding errors at play
+        try assertDecode(json: json, expected: proto)
+        try assertEncode(proto: proto, expected: json)
+    }
+
+    func testSmallDurationRoundtrip() throws {
+        let json = """
+        {
+          "duration":"1.000000010s",
```

Comment:
This might be an additional test, but I would test `0.1s` specifically, as it can't be represented by a `Float` or `Double`, so it would expose any floating point rounding that we're unintentionally doing.